packages/storage/src/internals/amplifyAuthConfigAdapter/getHighestPrecedenceUserGroup.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+export type UserGroupConfig = Record<string, Record<string, number>>[];
+
+/**
+ * Given a user group configuration and a list of current user groups,
+ * this function returns the user group with the highest precedence.
```

Comment:
Nit: this block describes only the shape of function input and output. I'd expect some explanation for how the highest precedence is determined. What about this?
```suggestion
 * Given the Cognito user groups associated to current user session
 * and all the user group configurations defined by backend. Find
 * the current session's user group that has highest precedence
 * according to backend configuration.
```